src/components/ColorPicker/ColorPicker.tsx

```diff
@@ -112,6 +141,15 @@ export class ColorPicker extends React.PureComponent<ColorPickerProps, State> {
     onChange({hue, brightness, saturation, alpha});
   };
 
+  private handleHexChange = (hex: string) => {
+    const {
+      color: {alpha = 1},
+      onChange,
+    } = this.props;
+    const newColor = hexToHsb(hex);
+    onChange({...newColor, alpha});
+  };
```

Comment:
Should we prevent invalid hex codes from being added, or maybe our error message could include why it's invalid? What do you think?

Reply:
This was discussed over in the [Figma designs](https://www.figma.com/file/yBEALRv5joQRqvyieFxmAl/Color-picker?node-id=4%3A2). We had error messages implemented first, but then decided to change to prevent invalid hex values being entered. The check is made in [`handleBlur`](https://github.com/Shopify/polaris-react/pull/2186/files#diff-b3d0546d98cde51214257201a3fb16adR104-R122) in the `TextPicker` component, by using the last valid value if an invalid value is entered.

Reply:
Interesting, so if a user enters `#1234567` they'll have to enter the code again rather than deleting the extra character?

Reply:
I'll leave this up to Miru's best judgment 😄